### PR TITLE
[WIP] Facility locator API initial checkin

### PR DESCRIPTION
**Pull request by patrickvinograd**
Still work in progress but wanted to get some early feedback before I over-polish anything. My rails background is pretty minimal so anything is fair game. 

Things I know are still needed:
- Rspec coverage. I'm still puzzling through that and vcr. 
- Handling of connection errors/timeouts from the GIS backend. Suggestions for patterns to follow appreciated.

Tag in any others whom you think should take a look.

### CODE REVIEW

**Comment by saneshark** [comment]
@patrickvinograd if you run `bundle exec rake` it will run all of the same scripts that travis does, including rubocop for linter. To fix a large swathe of rubocop errors, you can use `bundle exec rubocop -a` and it will correct the changes that it can.

As for writing specs, I would just look at existing ones and try to pickup those. One of the other things the rake task will check for is code coverage, it generates a report everytime your run the test suite, but to more granularly view what lines are missing coverage and write specs for those, you will want to open the index.html file in the coverage folder inside your web browser. 

As for handling exceptions, check out what some of the other services are doing in lib/<whatever> folder, namely evss, rx, and sm.

**Comment by saneshark** [comment]
For recording with VCR, you can copy and paste from existing specs, but you will need to specify a new path of your own. If the file does not exist at that path, it will be created when you run your specs the very first time. That being said, you will want to make sure that your local environment variables are set and you're connected to the VA network (if applicable), so that it can make the external requests properly. If for some reason you need to re-record, just delete the .yml file that is generated and run specs again. You can find more documentation on github.

File: app/models/va_health_facility.rb [hunk 1]
```diff
@@ -1,115 +1,116 @@
...
-    results.each_with_object([]) do | record, facs |
+    results.each_with_object([]) do |record, facs|
       facs << VAHealthFacility.from_gis(record)
     end
   end
 
   def self.find_by_id(id:)
     results = client.get(identifier: id)
-    VAHealthFacility.from_gis(results.first) unless results.nil?
+    VAHealthFacility.from_gis(results.first) unless results.nil? || results.empty?
```
Comment (bshyong) [inline app/models/va_health_facility.rb]: `blank?` is a Rails method that will combine these two 
http://api.rubyonrails.org/classes/Object.html#method-i-blank-3F

File: app/models/va_health_facility.rb [hunk 2]
```diff
@@ -14,7 +14,7 @@ def self.query(bbox:, services:)
...
```
Comment (bshyong) [inline app/models/va_health_facility.rb]: blank? is a Rails method that will combine these two 
http://api.rubyonrails.org/classes/Object.html#method-i-blank-3F

**Comment by patrickvinograd** [comment]
Closed, replaced by #162

File: lib/facilities/client.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+require 'geoservices'
+require 'openssl'
+
+# TODO: PV: Obviously this is not useful, but maps.va.gov has a self-signed cert
+# so try this for testing purposes.
+OpenSSL::SSL::VERIFY_PEER = OpenSSL::SSL::VERIFY_NONE
```
Comment (carlmjohnson) [inline lib/facilities/client.rb]: See https://github.com/department-of-veterans-affairs/vets-api/blob/master/lib/evss/base_service.rb#L74:L100 for what we did for EVSS's cert.

**Pull request closed without merging**